server_config.cc

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include "server_config.h"
+#include <string>
+
+ServerConfig::ServerConfig(NginxConfig config) {
+	fillOutMap(config);
+}
+
+ServerConfig::~ServerConfig() {
+  // TODO
+}
+
+void ServerConfig::fillOutMap(NginxConfig config) {
+	// Initialize variables property, value
+	std::string property = "";
+	std::string value = "";
+	for (size_t i = 0; i < config.statements_.size(); i++) {
+		//search in child block
+		if (config.statements_[i]->child_block_ != nullptr) {
+      fillOutMap(*(config.statements_[i]->child_block_));
+    }
+    property = config.statements_[i]->tokens_[0];
+    if (config.statements_[i]->child_block_ == nullptr) {
+      value = config.statements_[i]->tokens_[1];
+    }
+    if (property == "listen")
```

Comment:
It looks like this if-block has been mangled on its indentation and braces. Can you double-check this? Also, is there ever a case where `value` will not be set before we store it into our internal map?